unison-core/src/Unison/Name.hs

```diff
@@ -130,15 +122,16 @@ parent (Name txt) = case unsnoc (Text.splitOn "." txt) of
   Nothing -> Nothing
   Just ([],_) -> Nothing
   Just (init,_) -> Just $ Name (Text.intercalate "." init)
+parent (Name' names) =
+  Name' <$> NonEmpty.nonEmpty (NonEmpty.init names)
 
 suffixes :: Name -> [Name]
 suffixes (Name n) =
   fmap up . tails . dropWhile (== "") $ Text.splitOn "." n
   where
   up ns = Name (Text.intercalate "." ns)
-
-unqualified' :: Text -> Text
-unqualified' = last . Text.splitOn "."
+suffixes (Name' names) =
+  map (Name' . NonEmpty.fromList) (NonEmpty.init (NonEmpty.tails names))
```

Comment:
`NonEmpty.fromList` is safe here because `NonEmpty.init . NonEmpty.tails` contains no empty lists.